src/main/java/org/broadinstitute/dsm/util/DDPKitRequest.java

```diff
@@ -169,4 +198,100 @@ public void requestAndWriteKitRequests(List<LatestKitRequest> latestKitRequests)
             throw new RuntimeException("Error getting KitRequests ", e);
         }
     }
-}
+
+    public static void addOtherUnorderedKitsToList(Map<KitRequestSettings, ArrayList<KitRequest>> kitsToOrder) {
+        for (KitRequestSettings kitRequestSettings : kitsToOrder.keySet()) {
+            String query = "SELECT * FROM dev_dsm_db.ddp_kit_request request LEFT JOIN dev_dsm_db.ddp_kit kit on kit.dsm_kit_request_id = request.dsm_kit_request_id " +
+                    "LEFT JOIN ddp_kit_request_settings settings on settings.ddp_instance_id = request.ddp_instance_id " +
+                    "LEFT JOIN ddp_instance realm on (realm.ddp_instance_id = settings.ddp_instance_id) " +
+                    "where request.ddp_instance_id = ? and kit.needs_approval = true and kit.authorization = true and request.external_order_status is null";
+            ArrayList<KitRequest> orderKit = kitsToOrder.get(kitRequestSettings);
+            DDPInstance ddpInstance = DDPInstance.getDDPInstanceById(kitRequestSettings.getDdpInstanceId());
+            SimpleResult results = inTransaction((conn) -> {
+                SimpleResult dbVals = new SimpleResult();
+                try (PreparedStatement stmt = conn.prepareStatement(query)) {
+                    stmt.setInt(1, kitRequestSettings.getDdpInstanceId());
+                    try (ResultSet rs = stmt.executeQuery()) {
+                        while (rs.next()) {
+                            String participantsIndexES = ddpInstance.getParticipantIndexES();
+                            Map<String, Map<String, Object>> participantsESData = null;
+                            if (StringUtils.isNotBlank(participantsIndexES)) {
+                                //could be filtered as well to have a smaller list
+                                participantsESData = ElasticSearchUtil.getDDPParticipantsFromES(ddpInstance.getName(), participantsIndexES);
```

Comment:
please move that out of the while, so that you do not every time ask ES for the whole pt list. ask it ones after you ask for the DDPInstance and then get the pt you need out of the map